.github/workflows/build-and-deploy-site.yml

```diff
@@ -0,0 +1,53 @@
+name: Build and Deploy Site
+
+on:
+  push:
+    branches: [ "main", "html_page" ]
+
+jobs:
+  build-and-deploy-site:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout latest code
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+            distribution: zulu
+            java-version: 11
+
+      - name: Setup build cache
+        uses: pat-s/always-upload-cache@v2
+        with:
+            path: ~/.gradle/caches
+            key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+            restore-keys: |
+              ${{ runner.os }}-gradle-
+
+      - name: Set up ld cache
+        run: sudo ldconfig $(realpath libs)
+
+      - name: Check dependencies
+        run: |
+                ldd src/jvmMain/resources/lib/linux64/*.so
+      - name: Build project using Gradle
+        run: ./gradlew build -x test --stacktrace --scan
```

Comment:
Do you need to build and test a complete project just to deploy a site?
If yes, setup a dependency of this workflow on the "build" workflow, and deploy only if build has passed.
If no, then answer yourself additional question: "is it OK to deploy a site even when the rest of the project does not complile/pass the tests on CI"? If yes, then shrink this "deploy" workflow. If no, see the first passage.